internal/server/data/provideruser_test.go

```diff
@@ -254,3 +255,108 @@ func TestDeleteProviderUser(t *testing.T) {
 		assert.NilError(t, err)
 	})
 }
+
+func TestListProviderUsers(t *testing.T) {
+	type testCase struct {
+		name  string
+		setup func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser)
+	}
+
+	testCases := []testCase{
+		{
+			name: "list all provider users",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				pu := createTestProviderUser(t, tx, provider, "david@example.com")
```

Comment:
Not a blocker, but I've been trying to include a few records that I expect not to match the query in most of the tests for `List` operations. Something like same email different provider.